README.md

```diff
@@ -118,24 +159,28 @@ So, you can consider 1 as a good default value; it's strongly discouraged to set
 
 ### How the cache works
 
-`OwmCacheClient`'s internal cache follows this structure:
+`OwmCacheClient` and `DarkSkyCacheClient`'s internal cache stores responses in a map, which key is
+computed roughly as:
 
-|  Key                                                  |  Value            |
-| ----------------------------------------------------- | ----------------- |
-|  `round(latitude), round(longitude), beginningOfDay`  |  `List[Weather]`  |
+`round(latitude), round(longitude), beginningOfDay` 
 
 After the first request, it will try to fetch information for this exact place (round(latitude), round(longitude)) for the given day.
-This cache value usually consist of several (8-24) `Weather` objects, each with its own timestamps. `getCachedOrRequest` will return closest weather for specified timestamp and leave other in cache.
+The OWM cache value usually consists of several (8-24) `Weather` objects, each with its own timestamps.
 
+`cachingHistoryByCoords` in the OWM client will return the closest weather for the specified timestamp and leave the others in the cache.
 For example, the following code will make only one request to OpenWeatherMap, but return 3 distinct results:
 
 ```scala
-val client = OwmCacheClient[IO](MYKEY, 10, 1, 5.seconds)
-client.getCachedOrRequest(10.4f, 32.1f, 1514765952).unsafeRunSync()   // Jan 1 12:19:12 2018.
-client.getCachedOrRequest(10.1f, 32.312f, 1514765952).unsafeRunSync() // Jan 1 15:06:47 2015. From cache
-client.getCachedOrRequest(10.2f, 32.4f, 1514765952).unsafeRunSync()   // Nov 9 19:20:57 2015. From cache
+val client = OpenWeatherMap.cacheClient[IO]("YOUR_KEY", cacheSize = 10, geoPrecision = 1, timeout = 5.seconds)
```

Comment:
caching client creations returns an `F` so this doesn't compile.

also i think it'd be good to mention that specificity somewhere.